code/modules/clothing/glasses/glasses.dm

```diff
@@ -136,6 +136,13 @@
 	item_state = "glasses"
 	vision_correction = 1 //corrects nearsightedness
 
+/obj/item/clothing/glasses/regular/jamjar
+	name = "Jamjar Glasses"
+	desc = "Also known as Virginity Protectors."
+	icon_state = "jamjar_glasses"
+	item_state = "jamjar_glasses"
+	vision_correction = 1 //corrects nearsightedness

```

Comment:
@AnonymousNow your type is a CHILD of it's PARENT
```
child: /obj/item/clothing/glasses/regular/jamjar
parent: /obj/item/clothing/glasses/regular
```
and it INHERITs from it's PARENT all of it's PARENT's variables.